### PR TITLE
feat: add loader and submit button features

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-2510](https://openedx.atlassian.net/browse/PROD-2510)

This PR adds the following:

1. Shows a loader whenever submit button is pressed until the request is completed (also avoids multiple patch/post requests)
2. Submit button is hidden for all actions except create action AFTER the request has been completed
3. New Mode input updated. Now it doesn't include the active enrollment mode.
image

### CODE REVIEW

File: src/users/entitlements/v2/ExpireEntitlementForm.jsx [hunk 1]
```diff
@@ -102,18 +104,19 @@ export default function ExpireEntitlementForm({
...
+            <Button
+              variant="primary"
+              className="mr-3"
+              disabled={!(entitlement.courseUuid && entitlement.mode && comments)}
+              hidden={hideSubmit}
+              onClick={submit}
+            >
+              Submit
+            </Button>
+          ),
```
Comment (DawoudSheraz) [inline src/users/entitlements/v2/ExpireEntitlementForm.jsx:119]: Please add a relevant test for it or update an existing test to properly test this.

File: src/users/entitlements/v2/ReissueEntitlementForm.jsx [hunk 2]
```diff
@@ -100,18 +102,19 @@ export default function ReissueEntitlementForm({
...
+            <Button
+              variant="primary"
+              className="mr-3"
+              disabled={!(entitlement.courseUuid && entitlement.mode && comments)}
+              hidden={hideSubmit}
+              onClick={submit}
+            >
+              Submit
+            </Button>
+          ),
```
Comment (DawoudSheraz) [inline src/users/entitlements/v2/ReissueEntitlementForm.jsx:117]: same here.

**Review by DawoudSheraz — CHANGES_REQUESTED** [review]
to add missing unit tests.

**Review by azanbinzahid — APPROVED** [review]

**Review by DawoudSheraz — APPROVED** [review]
rebase before the merge.

**Pull request merged**